### PR TITLE
[GIT]  update .gitignore, ignore compdb

### PR DESCRIPTION
**Pull request by Simonhancrew**
update .gitignore, ignore compdb cache + compile_commands.json

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -66,3 +66,8 @@ libstandaloneengine.a
 tests/string
 tests/config
 
+# compdb
+.cache/
+compile_commands.json
+
+
```
Comment (vszakats) [inline .gitignore:73]:
```suggestion
```

Comment (vszakats) [inline .gitignore:73]: I suggest deleting the extra empty lines at the end of the file. (But keep the LF at the end of the last line)

Comment (Simonhancrew) [reply .gitignore]: Its done, thanks for your advice

**Comment by bagder** [comment]
Why should curl care about "combdb" in its ignore file?

**Comment by Simonhancrew** [comment]
I believe that many people use clangd as a code jumping and code hinting tool, which is why I wanted to add these. 

I think something that repo does not focus on should be divided into two parts, one is the compilation products, and the other is the dependency product of the code reading toolsets

**Comment by Simonhancrew** [comment]
anyway, its only my opinion, hh

**Comment by vszakats** [comment]
For non-curl-specific files, it seems better to ignore them in the global `.gitignore` that applies to all projects: `$XDG_CONFIG_HOME/git/ignore`.

**Comment by bagder** [comment]
> it seems better to ignore them via the global ignore list

Right. Files created by curl files are fine to ignore by curl ignore files, but files and leftovers generated by random other helper tools are better ignored using the generic global ignore. Otherwise there will be no end to what we need to ignore.

**Comment by Simonhancrew** [comment]
1, i got that,let me close this pr

**Pull request closed without merging**